src/gar.lua

```diff
@@ -187,21 +187,19 @@ function gar.leaveNetwork(from, currentTimestamp, msgId)
 
 	-- if the slash happens to be 100% we do not need to vault anything
 	if minimumStakedTokens > 0 then
-		gateway.vaults[from] = {
-			balance = minimumStakedTokens,
-			startTimestamp = currentTimestamp,
-			endTimestamp = gatewayEndTimestamp,
-		}
+		gar.createGatewayWithdrawVault(gateway, from, minimumStakedTokens, currentTimestamp, gatewayEndTimestamp)
```

Comment:
This should be a different function like "createGatewayExitVault" that centralizes the logic for when the end timestamp should be.

Reply:
[7ff44c4](https://github.com/ar-io/ar-io-network-process/pull/269/commits/7ff44c458ca916a481396b573ca7942ffb8cc3d8)